payment/controllers.go

```diff
@@ -0,0 +1,292 @@
+		if err != nil {
+			return handlers.WrapError(err, "Error retrieving the order", http.StatusInternalServerError)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		if err := json.NewEncoder(w).Encode(order); err != nil {
+			return handlers.WrapError(err, "Error encoding the orders JSON", http.StatusInternalServerError)
+		}
+		return nil
+	})
+}
+
+// CreateTransactionRequest includes information needed to create a transaction
+type CreateTransactionRequest struct {
+	ExternalTransactionID string `json:"externalTransactionID"`
+}
+
+// CreateTransaction creates a transaction against an order
+func CreateTransaction(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateTransactionRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		orderID := chi.URLParam(r, "orderID")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
+			return &handlers.AppError{
+				Message: "Error validating request url parameter",
+				Code:    http.StatusBadRequest,
+				Data: map[string]interface{}{
+					"validationErrors": map[string]string{
+						"orderID": "orderID must be a uuidv4",
+					},
+				},
+			}
+		}
+		validOrderID, err := uuid.FromString(orderID)
+		if err != nil {
+			panic(err) // Should not be possible
+		}
+
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		transaction, err := service.CreateTransactionFromRequest(req, validOrderID)
+		if err != nil {
+			return handlers.WrapError(err, "Error creating the transaction", http.StatusInternalServerError)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		if err := json.NewEncoder(w).Encode(transaction); err != nil {
+			return handlers.WrapError(err, "Error encoding the transaction JSON", http.StatusInternalServerError)
+		}
+		return nil
+	})
+}
+
+// FIXME should this be consollidated with the above?
+
+// CreateAnonCardTransactionRequest includes information needed to create a anon card transaction
+type CreateAnonCardTransactionRequest struct {
+	WalletID    uuid.UUID `json:"paymentId"`
+	Transaction string    `json:"transaction"`
+	Kind        string    `json:"kind"`
+}
+
+// CreateAnonCardTransaction creates a transaction against an order
+func CreateAnonCardTransaction(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateAnonCardTransactionRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		orderID := chi.URLParam(r, "id")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
+			return &handlers.AppError{
```

Comment:
same here. basically anywhere  we use an app error in a handler

Reply:
https://github.com/brave-intl/bat-go/pull/226/commits/9957ecb0f2690f79b044d501db73269ad8e501ac